config.h.in

```diff
@@ -0,0 +1,48 @@
+/* Copyright message. */
+#mesondefine COPYRIGHT
+/* Version number of package */
+#mesondefine VERSION
+/* Define if ancient vserver support enabled. */
+#mesondefine HAVE_ANCIENT_VSERVER
+/* Define if cgroup support enabled. */
+#mesondefine HAVE_CGROUP
+/* Define if delay accounting support should be enabled. */
+#mesondefine HAVE_DELAYACCT
+/* Define to 1 if you have the <hwloc.h> header file. */
+#mesondefine HAVE_HWLOC_H
+/* Define to 1 if you have the `hwloc' library (-lhwloc). */
+#mesondefine HAVE_LIBHWLOC
+/* Define to 1 if you have the `kstat' library (-lkstat). */
+#mesondefine HAVE_LIBKSTAT
+/* Define to 1 if you have the `kvm' library (-lkvm). */
+#mesondefine HAVE_LIBKVM
+/* The library is present. */
+#mesondefine HAVE_LIBNCURSESW
+/* Define to 1 if you have the <curses.h> header file. */
+#mesondefine HAVE_CURSES_H
+/* Define to 1 if you have the <ncursesw/curses.h> header file. */
+#mesondefine HAVE_NCURSESW_CURSES_H
+/* Define to 1 if you have the <ncurses/curses.h> header file. */
+#mesondefine HAVE_NCURSES_CURSES_H
+/* Define to 1 if you have the <ncurses.h> header file. */
+#mesondefine HAVE_NCURSES_H
+/* Define to 1 if you have the <ncurses/ncurses.h> header file. */
+#mesondefine HAVE_NCURSES_NCURSES_H
+/* Define to 1 if you have the `proc' library (-lproc). */
+#mesondefine HAVE_LIBPROC
+/* Define if Linux sched_setaffinity and sched_getaffinity are to be used. */
+#mesondefine HAVE_LINUX_AFFINITY
+/* Define if openvz support enabled. */
+#mesondefine HAVE_OPENVZ
+/* Define if setuid support should be enabled. */
+#mesondefine HAVE_SETUID_ENABLED
+/* Define if vserver support enabled. */
+#mesondefine HAVE_VSERVER
+/* Define to 1 if `major', `minor', and `makedev' are declared in <mkdev.h>.
+   */
+#mesondefine MAJOR_IN_MKDEV
+/* Define to 1 if `major', `minor', and `makedev' are declared in
+   <sys/sysmacros.h>. */
+#mesondefine MAJOR_IN_SYSMACROS
+/* Path of proc filesystem */
+#mesondefine PROCDIR
```

Comment:
If your `config.h.in` only contains `#mesondefine` statements, you can omit it entirely. See: https://mesonbuild.com/Reference-manual.html#configure_file

>  input: the input file name. If it's not specified in configuration mode, all the variables in the configuration: object (see above) are written to the output: file.

This is also less error-prone because you don't need to remember to edit two files when adding a new define.